kolibri/plugins/device_management/assets/src/modules/userPermissions/handlers.js

```diff
@@ -41,21 +47,39 @@ function fetchUserPermissions(userId) {
  * @returns Promise<void>
  */
 export function showUserPermissionsPage(store, userId) {
-  const promise = ConditionalPromise.all([
-    fetchUserPermissions(userId),
-    store.dispatch('getFacilities'),
-  ]).only(samePageCheckGenerator(store));
-  return promise
-    .then(function onUserSuccess([data]) {
-      return store.commit('userPermissions/SET_STATE', data);
+  const setAppBarTitle = title => store.commit('coreBase/SET_APP_BAR_TITLE', title);
+  const setUserPermissionsState = state => store.commit('userPermissions/SET_STATE', state);
+  const stopLoading = () => store.commit('CORE_SET_PAGE_LOADING', false);
+
+  // Don't request any data if not an Admin
+  if (!store.getters.isSuperuser) {
+    setUserPermissionsState({ user: null, permissions: {} });
+    setAppBarTitle(translator.$tr('goBackTitle'));
+    stopLoading();
+    return Promise.resolve();
+  }
+
+  // CoreBase parameters for loading state
+  setAppBarTitle(translator.$tr('loading'));
```

Comment:
Hmm I haven't seen this pattern anywhere. We can't display the loading page?

Reply:
It would still look weird. The loading page only fills the area below the App Bar, so it's helpful to have "Loading" text on the immersive toolbar. Otherwise it would visually change to something like this "<- Device", which I think looks odd.

Reply:
When I test this it's super fast, so the loading message flickers in and out very briefly:

![flicker](https://user-images.githubusercontent.com/2367265/49540139-27afbc00-f884-11e8-924d-16dd2b307573.gif)

Are there situations where the loading actually takes long enough to warrant a feedback indicator? (Unless the user actually needs it, I usually avoid trying to show loading feedback)

Reply:
created a follow-up issue for this: #4559